app/main.py

```diff
@@ -1,3 +1,49 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+
+from app.customer import Customer
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+    with open("app/config.json", "r") as f:
+        data = json.load(f)
+        fuel_price = data["FUEL_PRICE"]
+        for customer_data in data["customers"]:
+            customer = Customer(**customer_data)
+            print(f"{customer.name} has {customer.money} dollars")
+            closest = {}
+            shop_names = {}
+            for shop in Shop.create_shops("app/config.json"):
+                fuel = customer.car["fuel_consumption"]
+                expenses = (
+                    customer.product_total(shop.products)
+                    + customer.fuel_expenses(fuel, shop.location, fuel_price)
+                )
```

Comment:
And here do as follow
```suggestion
                if not shop. is_all_products_exist(customer_data.products_cart):
                    continue
                expenses = (
                    customer.product_total(shop.products)
                    + customer.fuel_expenses(fuel, shop.location, fuel_price)
                )
```

Reply:
after all changes:
<img width="876" alt="Screenshot 2023-03-06 at 18 11 41" src="https://user-images.githubusercontent.com/10571459/223166937-264f0d01-384e-4e8d-943b-760b67e3c58f.png">
<img width="670" alt="Screenshot 2023-03-06 at 18 13 16" src="https://user-images.githubusercontent.com/10571459/223167287-89a18d60-c566-4439-aaaa-e326157c545e.png">
<img width="633" alt="Screenshot 2023-03-06 at 18 13 35" src="https://user-images.githubusercontent.com/10571459/223167356-12fead20-5909-47bb-96f8-0234c954bcd8.png">

<img width="684" alt="Screenshot 2023-03-06 at 18 15 43" src="https://user-images.githubusercontent.com/10571459/223167938-24634dfc-9628-479b-bbc7-8f96303e9f74.png">

the result is:
<img width="768" alt="Screenshot 2023-03-06 at 18 12 42" src="https://user-images.githubusercontent.com/10571459/223167136-3844af44-34b8-4ba7-940e-375c4dd9cea8.png">